x/fungible/keeper/msg_server_update_zrc20_withdraw_fee.go

```diff
@@ -0,0 +1,84 @@
+package keeper
+
+import (
+	"context"
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/zeta-chain/protocol-contracts/pkg/contracts/zevm/zrc20.sol"
+	"github.com/zeta-chain/zetacore/x/fungible/types"
+	zetaObserverTypes "github.com/zeta-chain/zetacore/x/observer/types"
+)
+
+func (k Keeper) UpdateZRC20WithdrawFee(goCtx context.Context, msg *types.MsgUpdateZRC20WithdrawFee) (*types.MsgUpdateZRC20WithdrawFeeResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	if msg.Creator != k.zetaobserverKeeper.GetParams(ctx).GetAdminPolicyAccount(zetaObserverTypes.Policy_Type_deploy_fungible_coin) {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrUnauthorized, "Deploy can only be executed by the correct policy account")
+	}
+	zrc20Addr := ethcommon.HexToAddress(msg.Zrc20)
+	if zrc20Addr == (ethcommon.Address{}) {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid zrc20 contract address (%s)", msg.Zrc20)
+	}
+
+	// update contracts
+	zrc20ABI, err := zrc20.ZRC20MetaData.GetAbi()
+	if err != nil {
+		return nil, sdkerrors.Wrapf(types.ErrABIGet, "failed to get zrc20 abi")
+	}
+
+	foreignCoins := k.GetAllForeignCoins(ctx)
+	found := false
+	var coin types.ForeignCoins
+	for _, fcoin := range foreignCoins {
+		coinZRC20Addr := ethcommon.HexToAddress(fcoin.Zrc20ContractAddress)
+		if coinZRC20Addr == (ethcommon.Address{}) {
+			k.Logger(ctx).Error("invalid zrc20 contract address", "address", fcoin.Zrc20ContractAddress)
+			continue
+		}
+		if coinZRC20Addr == zrc20Addr {
+			coin = fcoin
+			found = true
+			break
+		}
+	}
+
+	if !found {
+		return nil, sdkerrors.Wrapf(types.ErrInvalidAddress, "no foreign coin match requested zrc20 address (%s)", msg.Zrc20)
+	}
```

Comment:
Is there a reason of not using
```
GetForeignCoins(
	ctx sdk.Context,
	zrc20Addr string,
)
```
to directly get the coin from its address?

Reply:
i think we might have a problem of normalization of eth address. 
GetForeignCoins() needs exact zrc20Addr, here we can use all lowercase or whatever is convenient.

Reply:
Shouldn't we consider indexing the foreign coin by the Ethereum address? There might be gas efficiency issue in the future with the iteration if we have a lot of coins?

Reply:
yeah we could do that.

Reply:
https://github.com/zeta-chain/node/issues/948